### PR TITLE
Implement main / handler pattern for more examples

### LINKED ISSUE #1679: missing instructions on how to update the private set intersection example when the Wasm module is modified
Not sure if actually missing or I am looking in the wrong place.

@ipetr0v do you know if we have anything like this? Or do people just do it from scratch every time?

### PR DESCRIPTION
**Pull request by tiziano88**
Fix #1679

# Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by
        [Cloudbuild](/cloudbuild.yaml)
  - [ ] I have updated [documentation](/docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: examples/private_set_intersection/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Private Set Intersection
```
Comment (ipetr0v) [inline examples/private_set_intersection/README.md:1]: Should we move these instructions to `examples/README.md` so they could apply to all examples?

File: examples/private_set_intersection/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+1. Build the example, including the Wasm module
+1. Push the module to GS via
+   `./scripts/push-example -e private_set_intersection`
+1. Fix the URL and hash in [`./oak_app_manifest.toml`](./oak_app_manifest.toml)
+1. Sign the module with
+
+   ```bash
+   ./scripts/oak_sign sign \
+     --private-key=examples/keys/ed25519/test.key \
```
Comment (ipetr0v) [inline examples/private_set_intersection/README.md:13]: I think we need an explicit comment stating that here we use a Test key that is only used for debugging

**Review by ipetr0v — APPROVED** [review]

File: examples/private_set_intersection/README.md (hunk 1 above)
Comment (tiziano88) [reply examples/private_set_intersection/README.md:1]: I think it needs more work to make them applicable to the other examples (do any other examples actually rely on signatures and uploading modules?), perhaps let's do it in a separate PR.

File: examples/private_set_intersection/README.md (hunk 2 above)
Comment (tiziano88) [reply examples/private_set_intersection/README.md:13]: Done.

File: examples/private_set_intersection/README.md (hunk 1 above)
Comment (ipetr0v) [reply examples/private_set_intersection/README.md:1]: Agree
Currently we only have Aggregator and Private Set Intersection that use remotely downloaded modules

**Pull request merged**